### My Question
Does a person's appetite for risk (measured by their choice between a risky bet, Lottery A, and a safe bet, Lottery B) relate to how they prefer their steak cooked (Rare, Medium rare, Medium, Medium Well, or Well)?

### My Study
This is an **observational study**.
- It is **not an experiment** because nobody was assigned a treatment or forced into a group. The researchers just collected survey responses about people's existing habits and preferences.
- It is **not a census** because it didn't collect data from every adult in the US. It was an online survey of 550 people through SurveyMonkey.

### My Analysis

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

sns.set_theme(style="whitegrid")

# Load dataset and skip the leftover survey sub-header row
df = pd.read_csv("steak.csv", skiprows=[1])

# Rename columns to make them easier to work with
df.columns = [
    "id", "lottery", "smoke", "alcohol", "gamble", "skydiving",
    "speeding", "cheated", "eat_steak", "steak_doneness",
    "gender", "age", "income", "education", "location"
]

# Drop rows where steak doneness or lottery choice is missing
df_clean = df.dropna(subset=["steak_doneness", "lottery"]).copy()

# Put steak doneness in order
doneness_order = ["Rare", "Medium rare", "Medium", "Medium Well", "Well"]
df_clean["steak_doneness"] = pd.Categorical(df_clean["steak_doneness"], categories=doneness_order)

# 1. Frequency Table (Raw counts)
freq_table = pd.crosstab(df_clean["steak_doneness"], df_clean["lottery"])[["Lottery A", "Lottery B"]]
print("--- Frequency Table (Raw Counts) ---")
print(freq_table)

# 2. Relative Frequency Table (Percentages)
rel_pct = (pd.crosstab(df_clean["steak_doneness"], df_clean["lottery"], normalize="columns") * 100)[["Lottery A", "Lottery B"]].round(1)
print("\n--- Relative Frequency Table (%) ---")
print(rel_pct)

# 3. Reshape for Seaborn
rel_long = rel_pct.reset_index().melt(id_vars="steak_doneness", var_name="lottery", value_name="percent")

# 4. Grouped Bar Chart
plt.figure(figsize=(9, 5))
ax = sns.barplot(
    data=rel_long,
    x="steak_doneness",
    y="percent",
    hue="lottery",
    palette=["#4C72B0", "#DD8452"],
    edgecolor="black"
)
ax.set_title("Steak Doneness Preference by Risk Appetite (Lottery Choice)")
ax.set_xlabel("Steak Doneness")
ax.set_ylabel("Percent of Group (%)")
ax.legend(title="Lottery Bet")
plt.tight_layout()
plt.savefig("steak_lottery_comparison.png", dpi=150)
plt.show()


--- Frequency Table (Raw Counts) ---
lottery         Lottery A  Lottery B
steak_doneness                      
Rare                   10         13
Medium rare            73         91
Medium                 74         57
Medium Well            34         40
Well                   16         20

--- Relative Frequency Table (%) ---
lottery         Lottery A  Lottery B
steak_doneness                      
Rare                  4.8        5.9
Medium rare          35.3       41.2
Medium               35.7       25.8
Medium Well          16.4       18.1
Well                  7.7        9.0


### My Conclusion
1. **Answer & Key Statistics:**
- **Mode:** For risk-takers who picked Lottery A, the mode is **Medium** at **35.7%** (74 people), right alongside **Medium rare** at **35.3%** (73 people). For conservative betters who picked Lottery B, the clear mode is **Medium rare** at **41.2%** (91 people).
- **Relative Frequencies:** A higher share of Lottery A betters preferred their steak **Medium** (**35.7%**) compared to Lottery B betters (**25.8%**). On the other hand, Lottery B betters had a higher share choosing **Medium rare** (**41.2%** vs. **35.3%**).
- **Extreme Categories:** Both **Rare** (4.8% for A vs. 5.9% for B) and **Well** (7.7% for A vs. 9.0% for B) make up tiny, almost identical percentages across both groups. This shows that taking big risks on a bet doesn't mean you're more likely to eat raw or charred steak.

2. **Screenshot of Graph:**

![Steak Doneness by Lottery Choice](steak_lottery_comparison.png)

3. **Graph Interpretation:**
- Looking at the grouped bar chart, the distributions look pretty similar across both groups instead of showing a massive shift.
- The bars for Rare and Well done stay under 10% for both groups, meaning extreme doneness is rare regardless of risk appetite.
- The main difference is in the middle: the Lottery A bar is noticeably taller at Medium, while the Lottery B bar is taller at Medium rare. Overall, risk appetite doesn't really push people toward eating rare steak—most people in both groups just stick to Medium rare or Medium.